enhancements/20250407-cpu-model-features-selection.md

```diff
@@ -0,0 +1,385 @@
+- Force migration. It should be done [by upstream](https://github.com/kubevirt/kubevirt/issues/15623).
+
+## Introduction
+
+Before this enhancement, we need to understand how the CPU model and features work in KubeVirt. To make the CPU model and migration logic even easier to grasp, let's use a visual analogy with colored shapes:
+
+Imagine you have colored shapes representing CPU configurations:
+
+- Shapes: Represent CPU models (e.g., Circle, Square, Triangle).
+- Colors: Represent CPU features (e.g., Red, Blue, Green).
+
+Here's how they interact:
+
+- Scenario 1: Person requiring a specific color  
+    A person requires an object with a specific color (e.g., Red).  
+    You can give the person any shape that is red (e.g., a Red Square, a Red Circle, or a Red-Blue Circle).
+    However, red color might be only provided with a specific shape. It also means that a feature might be only provided with a specific CPU model.
+
+
+- Scenario 2: Person requiring a specific shape  
+  A person requires an object with a specific shape (e.g., Circle).  
+  You can give the person any circle, regardless of its color (e.g., a Red Circle, a Blue Circle, or a Red-Blue Circle).
+
+- Scenario 3: Person requiring a specific color and shape  
+  A person requires an object with a specific color and shape (e.g., Red Circle).  
+  You must give the person an object that matches the color and shape (e.g., a Red Circle or a Red-Blue Circle).
+
+- Scenario 4: Person requiring a moving to a new location  
+  A person requires an object with a specific color and shape (e.g., Red Circle) and is moving to a new location.  
+  The new location must provide an object that matches the color and shape (e.g., a Red Circle or a Red-Blue Circle).
+
+A node can provide objects with different shapes and colors. For example, a Node might provide a "Red Square", a "Blue Circle", and a "Red-Blue Circle". 
+
+A VM is like a person who requires different objects. 
+
+Migration is like moving a person with colored shape requirements to a new location that can fulfill those requirements. The destination must provide objects that meet the person's color and shape needs.
+
+You can treat shapes as `cpu-model.node.kubevirt.io/{cpu model}` and colors as `cpu-feature.node.kubevirt.io/{cpu feature}`. For more details, please check the Node section of this enhancement.
+
+## Proposal
+
+Due to multiple nodes, we can't show a big matrix for all the CPU models and features. Instead, we'll provide a dropdown selection menu with colored mark.
+
+### User Stories
+
+#### Story 1
+
+I have multiple nodes that have a common CPU model called `Nehalem`. For some reason, I'll need to migrate my VMs to other nodes without manually shutting down. To ensure compatibility, I create my VMs with `Nehalem` as the CPU model.
+
+### User Experience In Detail
+
+In general, there are two ways to migrate VMs between nodes.
+
+- Use a common CPU model
+- Use a Harvester special defined annotation
```

Comment:
What's the annotation?

Reply:
This line should be removed because we don't need force migration.